lib/cassandrax/keyspace/schema.ex

```
@@ -63,9 +72,13 @@ defmodule Cassandrax.Keyspace.Schema do
     {statement, values, changeset} = setup_insert(keyspace, changeset)
     {:ok, prepared} = Cassandrax.Connection.prepare(conn, statement)
 
-    case Cassandrax.Connection.execute(conn, prepared, values, opts) do
-      {:ok, _void_response} -> load_changes(changeset, :loaded)
-      {:error, error} -> {:error, error}
+    try do
+      case Cassandrax.Connection.execute(conn, prepared, values, opts) do
+        {:ok, _void_response} -> load_changes(changeset, :loaded)
+        {:error, error} -> {:error, error}
+      end
+    rescue
+      _ in FunctionClauseError -> {:error, :invalid_data}
```

I think you can leverage the error raised here (with its inner context) and return it, instead of returning the `:invalid_data` atom and then raising the `InvalidDataError` without context information from the caller.

---

ok i bubbled it up.